hopsworks-common/src/main/java/io/hops/hopsworks/common/provenance/core/elastic/ProvElasticController.java

```diff
@@ -0,0 +1,364 @@
+    throws E, ElasticException {
+    GetResponse response;
+    try {
+      LOG.log(Level.FINE, "request:{0}", request.toString());
+      response = client.getClient().get(request, RequestOptions.DEFAULT);
+    } catch (IOException e) {
+      String msg =  "error during get doc:" + request.id();
+      ElasticException se = processException(e, msg);
+      throw se;
+    }
+    if(response.isExists()) {
+      return resultParser.apply(BasicElasticHit.instance(response));
+    } else {
+      return null;
+    }
+  }
+  
+  public void indexDoc(IndexRequest request) throws ElasticException {
+    IndexResponse response;
+    
+    try {
+      LOG.log(Level.FINE, "request:{0}", request.toString());
+      response = client.getClient().index(request, RequestOptions.DEFAULT);
+    } catch (IOException e) {
+      String msg = "error during index doc:" + request.id();
+      ElasticException se = processException(e, msg);
+      throw se;
+    }
+    if (response.status().getStatus() != 201) {
+      String msg = "doc index - bad status response:" + response.status().getStatus();
+      LOG.log(Level.INFO, msg);
+      throw new ElasticException(RESTCodes.ElasticErrorCode.ELASTIC_QUERY_ERROR, Level.INFO, msg);
+    }
+  }
+  
+  public void updateDoc(UpdateRequest request) throws ElasticException {
+    UpdateResponse response;
+    
+    try {
+      LOG.log(Level.FINE, "request:{0}", request.toString());
+      response = client.getClient().update(request, RequestOptions.DEFAULT);
+    } catch (IOException e) {
+      String msg = "error during update doc:" + request.id();
+      ElasticException se = processException(e, msg);
+      throw se;
+    }
+    if (response.status().getStatus() != 200) {
+      String msg = "doc update - bad status response:" + response.status().getStatus();
+      LOG.log(Level.INFO, msg);
+      throw new ElasticException(RESTCodes.ElasticErrorCode.ELASTIC_QUERY_ERROR, Level.INFO, msg);
+    }
+  }
+  
+  public <S, A, E extends Exception> Pair<Long, S> search(SearchRequest request,
+    ElasticHitsHandler<?, S, A, E> resultParser) throws ElasticException, E {
+    SearchResponse response;
+    LOG.log(Level.FINE, "request:{0}", request.toString());
+    response = searchBasicInt(request);
+    resultParser.apply(response.getHits().getHits());
+    return Pair.with(response.getHits().getTotalHits().value, resultParser.get());
+  }
+  
+  public <S, A, E extends Exception> Pair<Long, S> searchScrolling(SearchRequest request,
+    ElasticHitsHandler<?, S, A, E> resultParser) throws ElasticException, E {
+    SearchResponse response;
+    long leftover;
+    LOG.log(Level.FINE, "request:{0}", request.toString());
+    response = searchBasicInt(request);
+    
+    if(response.getHits().getTotalHits().value > settings.getElasticMaxScrollPageSize()) {
+      String msg = "Elasticsearch query items size is too big: " + response.getHits().getTotalHits();
+      LOG.log(Level.INFO, msg);
```

Comment:
this will log the message twice. first when you call log and then when you throw the exception

Reply:
fixed